### DISCUSSION TITLE
Caching multiple files in one step (by iterating through an array?)

### DISCUSSION THREAD

**Discussion by hbasaric** [actions/cache#853] [Q&A]
Hello,

I am looking to create a cache key for a list of 37 files. I am trying to avoid having multiple blocks of code that create a cache key for each file individually. Is there a way to iterate over a list of files and create a key for each file? 

This is what my code looks like to create a cache key and install/cache a single file (FileName) within my GitHub Actions workflow run: 

     - name: Cache and install dependencies
        uses: actions/cache@v3
        id: cache
        with:
          path: ~/
          key: ${{ runner.os }}-java-FileName
          restore-keys: |
            ${{ runner.os }}-java-
            
      - if: steps.cache.outputs.cache-hit != 'true'
        name: Install deps
        run: |
          if [ ! -e ./FileName]; then
            wget downloadLinkForFileName
          fi

I would like to avoid repeating this block of code 37 times. If possible, I would like to define an array or a list of strings that correspond to each file name and iterate over this list to create a key for each file.

Thank you.

**Comment by jongwooo** [comment]
@hbasaric How about caching directories that contain 37 files?
